Docker builds fail

If the Greengrass Docker container has already been download the `-c` works as expected.

If it hasn't the user sees this error:

```
[INFO] BasicProgressHandler: Stream: Step 1/11 : FROM 216483018798.dkr.ecr.us-west-2.amazonaws.com/aws-iot-greengrass:latest
[INFO] BasicProgressHandler: Stream:

[ERROR] BasicProgressHandler: Docker build error [Get https://216483018798.dkr.ecr.us-west-2.amazonaws.com/v2/aws-iot-greengrass/manifests/latest: no basic auth credentials]
[ERROR] AwsGreengrassProvisioner: Could not acquire image ID or digest following build
```

--- a/src/main/java/com/awslabs/aws/greengrass/provisioner/implementations/helpers/BasicDeploymentHelper.java
+++ b/src/main/java/com/awslabs/aws/greengrass/provisioner/implementations/helpers/BasicDeploymentHelper.java
@@ -857,6 +857,10 @@ public class BasicDeploymentHelper implements DeploymentHelper {
             try (DockerClient dockerClient = officialGreengrassImageDockerClientProvider.get()) {
                 log.info("Building container");
 
+                // Pull the official Greengrass Docker container first
+                String officialGreengrassDockerImage = ggConstants.getOfficialGreengrassDockerImage();
+                officialGreengrassImageDockerHelper.pullImage(officialGreengrassDockerImage);
+
                 String imageId = dockerClient.build(new File(currentDirectory).toPath(),
                         basicProgressHandler,
                         DockerClient.BuildParam.dockerfile(tempDockerfile.toPath()));

